src/ParseFile.js

```diff
@@ -317,25 +317,27 @@ const DefaultController = {
     if (XHR) {
       return this.downloadAjax(uri);
     }
-    if (process.env.PARSE_BUILD === 'browser') {
+    if (process.env.PARSE_BUILD === 'browser' && !XHR) {
       return Promise.reject('Cannot make a request: No definition of XMLHttpRequest was found.');
     }
-    return new Promise((resolve, reject) => {
-      const client = uri.indexOf('https') === 0
-        ? require('https')
-        : require('http');
-      client.get(uri, (resp) => {
-        resp.setEncoding('base64');
-        let base64 = '';
-        resp.on('data', (data) => base64 += data);
-        resp.on('end', () => {
-          resolve({
-            base64,
-            contentType: resp.headers['content-type'],
+    if (process.env.PARSE_BUILD === 'node') {
```

Comment:
is it possilbe for this if to be false and neither of the two ifs above to be true?

if so, how to handle?  if not, then we shouldn't need this test?

Reply:
The first if handles react-native and browser
The second for unsupported browsers
The third is for node only. (Note babel trick here. This won't end up in the react-native build)

You can check by forcing this branch. `npm run build` and check `/lib/xxx/ParseFile.js` and notice the difference